### PR TITLE
Contoso Pizza Scenario

### PR DESCRIPTION
**Pull request by aforank**
image

### CODE REVIEW

**Comment by aforank** [comment]
Do not merge

File: Controllers/PizzaController.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        {
+        }
+
+        /// <summary>
+        /// Gets all pizzas.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        public ActionResult<List<Pizza>> GetAll() =>
+            PizzaService.GetAll();
```
Comment (pragatgo) [inline Controllers/PizzaController.cs:31]: PizzaService Class should have an interface

File: Services/PizzaService.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+using ContosoPizza.Models;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ContosoPizza.Services
+{
+    /// <summary>
+    /// Pizza Service.
+    /// </summary>
+    public static class PizzaService
```
Comment (pragatgo) [inline Services/PizzaService.cs:10]: Can we impalement the PizzaServcie class with interface and we can use dependency injection to inject the dependencies in other classes.

File: Controllers/PizzaController.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    /// <seealso cref="Microsoft.AspNetCore.Mvc.ControllerBase" />
+    [ApiController]
+    [Route("[controller]")]
+    public class PizzaController : ControllerBase
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PizzaController"/> class.
+        /// </summary>
+        public PizzaController()
+        {
```
Comment (pragatgo) [inline Controllers/PizzaController.cs:22]: Adding dependency of Pizza Service by an interface would be a better option.

File: Controllers/PizzaController.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+namespace ContosoPizza.Controllers
+{
+    /// <summary>
+    /// Pizza Controller.
+    /// </summary>
+    /// <seealso cref="Microsoft.AspNetCore.Mvc.ControllerBase" />
+    [ApiController]
+    [Route("[controller]")]
+    public class PizzaController : ControllerBase
```
Comment (pragatgo) [inline Controllers/PizzaController.cs:16]: Will be good to have unit test cases to verify the functionality.

File: WeatherForecast.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
+using System;
+
+namespace ContosoPizza
+{
+    /// <summary>
+    /// Weather Forecast class.
+    /// </summary>
+    public class WeatherForecast
```
Comment (pragatgo) [inline WeatherForecast.cs:8]: Can we move this file to model folder.

File: Controllers/WeatherForecastController.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+using Microsoft.Extensions.Logging;
+
+namespace ContosoPizza.Controllers
+{
+    /// <summary>
+    /// Weather Forecast Controller.
+    /// </summary>
+    /// <seealso cref="Microsoft.AspNetCore.Mvc.ControllerBase" />
+    [ApiController]
+    [Route("[controller]")]
```
Comment (pragatgo) [inline Controllers/WeatherForecastController.cs:15]: Can we have separate service file for business logic for reuse purposes?

File: Controllers/PizzaController.cs [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        /// Initializes a new instance of the <see cref="PizzaController"/> class.
+        /// </summary>
+        public PizzaController()
+        {
+        }
+
+        /// <summary>
+        /// Gets all pizzas.
+        /// </summary>
+        /// <returns></returns>
```
Comment (pragatgo) [inline Controllers/PizzaController.cs:28]: Can we please return the appropriate HTTP status code (Success and failure)

File: Controllers/PizzaController.cs [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        /// Gets the specified pizza by identifier.
+        /// </summary>
+        /// <param name="id">The identifier.</param>
+        /// <returns></returns>
+        [HttpGet("{id}")]
+        public ActionResult<Pizza> Get(int id)
+        {
+            var pizza = PizzaService.Get(id);
+
+            if (pizza == null)
```
Comment (pragatgo) [inline Controllers/PizzaController.cs:43]: Can we please return the appropriate status code for Null handling situation  (something like No Pizza Found)

File: Controllers/PizzaController.cs [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+        /// <summary>
+        /// Creates the specified pizza.
+        /// </summary>
+        /// <param name="pizza">The pizza.</param>
+        /// <returns></returns>
+        [HttpPost]
+        public IActionResult Create(Pizza pizza)
+        {
+            PizzaService.Add(pizza);
```
Comment (pragatgo) [inline Controllers/PizzaController.cs:57]: Can we have input validation?

**Pull request closed without merging**